src/main/java/jp/ikedam/jenkins/plugins/extensible_choice_parameter/DatabaseChoiceListProvider.java

```diff
@@ -0,0 +1,419 @@
+            {
+                return FormValidation.error(Messages.DbChoiceListProvider_jdbcDriver_empty());
+            }
+                       
+            return FormValidation.ok();
+        }
+        
+    }
+	
+	
+	/**
+	 * Recupere la liste des environnements recuperer de la base de donnees JACK	  
+	 *	 
+	 */
+	/**
+	 * 
+	 * @return
+	 */
+	private List<String> performDbQuery(){
+
+		Connection conn = null;
+		ResultSet resultat = null;
+		List<String> resultList = new ArrayList<String>();
+		try {
+			String driver = getJdbcDriver();
+			if (driver == null || StringUtils.isEmpty(driver)) {
+				LOGGER.log(Level.WARNING,"Invalid driver");
+			}
+			
+			Class.forName(driver);
+			
+			String user = getDbUsername();
+			if (user == null || StringUtils.isEmpty(user)) {
+				LOGGER.log(Level.WARNING,"Invalid user");
+			}
+
+			String password = getDbPassword();
+			if (password == null || StringUtils.isEmpty(password)) {
+				LOGGER.log(Level.WARNING,"Invalid password");
+			}
+
+			String urlBase = getJdbcUrl();
+			if (urlBase == null || StringUtils.isEmpty(urlBase)) {
+				LOGGER.log(Level.WARNING,"Invalid JDBC URL");				
+			}
+			
+			String colomndb = getDbColumn();
+			if (colomndb == null || StringUtils.isEmpty(colomndb)) {
+				LOGGER.log(Level.WARNING,"Invalid column");
+			}
+			
+			String namedb = getDbName();
+			if (namedb == null || StringUtils.isEmpty(namedb)) {
+				LOGGER.log(Level.WARNING,"Invalid DB name");
+			}
+
+			String tabledb = getDbTable();
+			if (tabledb == null || StringUtils.isEmpty(tabledb)) {
+				LOGGER.log(Level.WARNING,"table database invalid");				
+			}
+			
+			conn = DriverManager.getConnection(urlBase, user, password);
+						
+			// By default, a SELECT * is performed
+			String selectQuery = "select " + colomndb + " from " + namedb + "." + tabledb;
+						
+			// Use plain old JDBC to build and execute the query against the configured db
+			Statement statement = conn.createStatement();
+			boolean result = statement.execute(selectQuery);
+			if(result){
+				resultat = statement.executeQuery(selectQuery);
+				while(resultat.next()) {
+					resultList.add(resultat.getString(1));				  
+				}
+			}else{
+				LOGGER.log(Level.WARNING,"No result found with the query: " + selectQuery);
+			}
+		}
+		catch (SQLException se) {
+			LOGGER.log(Level.SEVERE,"Unable to access the database: " + dbName + "." + se);			
+		}
+		catch(ClassNotFoundException e) {
+			LOGGER.log(Level.SEVERE, "The driver " + jdbcDriver + " cannot be found in the classpath.");
+		}
+		catch (Exception e) {
+			LOGGER.log(Level.SEVERE,"Unable to access the database: " + dbName + "." +  e);
+		}
+		finally {
+			try {
+				if (conn != null) {
+					conn.close();
+				}
+			}
+			catch (SQLException se) {
+				LOGGER.log(Level.SEVERE,"Fermeture de connection impossible, SQLException : " + se);
+			}
+		}
+		
+		// If no results are returned, read values from the fallback file if configured
```

Comment:
This feature can be a vulnerability.
E.g. `fallbackFilepath="/etc/passwd"`

Yes, `FilenameChoiceListProvider` also can be a security vulnerability.